### PR TITLE
Intelligent term references and formatting

### PR DESCRIPTION
**Pull request by w00lf**
#255 

New macro: `term[X,Y]`, reference term definition throughout the document.

### CODE REVIEW

File: lib/asciidoctor/iso/cleanup.rb [hunk 1]
```diff
@@ -86,6 +88,42 @@ def sort_biblio(bib)
...
+          next if AUTOMATIC_GENERATED_ID_REGEXP.match(term_node['id']).nil?
+
+          term_text = term_node.at('./preferred').text
+          term_node['id'] = unique_text_id(xmldoc, term_text)
+          termlookup[term_text] = term_node['id']
+        end
+      end
+
+      def unique_text_id(xmldoc, text)
+        return "text-#{text}" if xmldoc.at("//*[@id = 'text-#{text}']").nil?
```
Comment (opoudjis) [inline lib/asciidoctor/iso/cleanup.rb:118]: "text-#{text}" should be "term-#{text}" throughout this method

File: lib/asciidoctor/iso/cleanup.rb [hunk 2]
```diff
@@ -86,6 +88,42 @@ def sort_biblio(bib)
...
+      def termdef_cleanup(xmldoc)
+        generate_termref_links(xmldoc)
+        super
+      end
+
+      def generate_termref_links(xmldoc)
+        termlookup = replace_automatic_genrated_ids_terms(xmldoc)
+        xmldoc.xpath('//termxref').each do |node|
+          target = node.text
+          next if termlookup[target].nil?
```
Comment (opoudjis) [inline lib/asciidoctor/iso/cleanup.rb:100]: Please log an error here, using `@log.add("AsciiDoc Input", node, "#{target} does not refer to a real term")`

Comment (w00lf) [reply lib/asciidoctor/iso/cleanup.rb]: Done

File: lib/asciidoctor/iso/cleanup.rb (hunk 1 above)
Comment (w00lf) [reply lib/asciidoctor/iso/cleanup.rb]: Done

**Comment by ronaldtse** [comment]
@w00lf @opoudjis why is this only in ISO?

**Comment by ronaldtse** [comment]
I think we should also have `term:[X]` as shorthand as `term:[X,X]`.

**Comment by ronaldtse** [comment]
@w00lf A PR ought to include documentation on how to use the function. We can't merge unless there is documentation. Thanks.

**Comment by ronaldtse** [comment]
@w00lf this PR is broken:

HTML:
image

PDF:
image

**Comment by w00lf** [comment]
> I think we should also have `term:[X]` as shorthand as `term:[X,X]`.

Currently, it does support such notataion

**Pull request merged**